spec/unit/knife/bootstrap_spec.rb

```diff
@@ -40,8 +40,8 @@
   let(:knife) do
     Chef::Log.logger = Logger.new(StringIO.new)
     Chef::Config[:knife][:bootstrap_template] = bootstrap_template unless bootstrap_template.nil?
-    expect(LicenseAcceptance::Acceptor).to receive(:check_and_persist!)
 
+    expect_any_instance_of(Chef::Knife::Bootstrap).to receive(:check_license)
     k = Chef::Knife::Bootstrap.new(bootstrap_cli_options)
```

Comment:
this is likely smell that you shouldn't be doing `chef_license` in the initializer, and should move that to `#run` (and really the ivars that are set in the initializer should be moved to lazy accessors so that the initializer disappears.  then you avoid having the testing against the initializer below, which reads really awkward, and that should eliminate the need to `expect_any_instance_of` and you can just set the expectation directly against `k` here.